metsim/methods/mtclim.py

```diff
@@ -26,74 +21,61 @@ def run(forcing: pd.DataFrame, disagg=True):
     lat_idx = forcing.index.names.index('lat')
     time_idx = forcing.index.names.index('time')
     sg = solar_geom(forcing['elev'][0], forcing.index.levels[lat_idx][0])
+    sg = {'tiny_rad_fract' : sg[0],
+          'daylength' : sg[1],
+          'potrad' : sg[2],
+          'tt_max0' : sg[3]}
     forcing.index = forcing.index.levels[time_idx]
-    calc_t_air(forcing)
-    calc_precip(forcing)
-    calc_snowpack(forcing)
-    calc_srad_hum(forcing, sg)
-    calc_longwave(forcing)
+    params['n_days'] = len(forcing.index)
+    calc_t_air(forcing, params)
+    calc_precip(forcing, params)
+    calc_snowpack(forcing, params)
+    calc_srad_hum(forcing, sg, params)
     
     if disagg:
-        forcing = disaggregate(forcing, sg)
+        forcing = disaggregate(forcing, params, sg)
 
     return forcing
 
 
-def calc_t_air(df: pd.DataFrame):
+def calc_t_air(df: pd.DataFrame, params: dict):
     """ 
     Adjust temperatures according to lapse rates 
     and calculate t_day
     """
-    dZ = (params['site_elev'] - params['base_elev'])/1000.
+    dZ = (df['elev'][0] - params['base_elev'])/1000.
     lapse_rates = [params['t_min_lr'], params['t_max_lr']]
-    df['t_max'] = df['t_max'] + dZ * lapse_rates[1]
-    df['t_min'] = np.minimum(df['t_min'] + dZ * lapse_rates[0], df['t_max']-0.5)
-    t_mean = np.mean(df['t_min'] + df['t_max'])
-    df['t_day'] = ((df['t_max'] - t_mean) * params['TDAYCOEF']) + t_mean
+    t_max = df['t_max'] + dZ * lapse_rates[1]
+    t_min = np.minimum(df['t_min'] + dZ * lapse_rates[0], t_max-0.5)
+    t_mean = (t_min + t_max)/2
```

Comment:
The call to `np.mean` collapses that down to a single value unless you provide it an `axis` argument.  I think this is more clear, and less prone to error in the future, while it doesn't affect performance in any significant (or measurable) way.  This was a bug that eluded me for quite a while, unfortunately.